### PR TITLE
feat: implement inscription detail and list page

### PR DESCRIPTION
**Pull request by WhiteMinds**
Refs:
https://github.com/Magickbase/ckb-explorer-public-issues/issues/517
https://github.com/Magickbase/ckb-explorer-public-issues/issues/518

### CODE REVIEW

File: src/models/UDT/index.ts [hunk 1]
```diff
@@ -17,3 +17,20 @@ export interface UDT {
...
   uan?: string
 }
+
+export enum MintStatus {
+  Minting = 'minting',
+  Closed = 'closed',
+  RebaseStart = 'rebase_start',
+}
+
+export interface OmigaInscription extends UDT {
```
Comment (Keith-CY) [inline src/models/UDT/index.ts:27]: I would prefer naming it clearly to distinguish `inscription collection` and `inscription item`

File: src/models/UDT/index.ts [hunk 2]
```diff
@@ -17,3 +17,20 @@ export interface UDT {
...
+  mintStatus: MintStatus
+  mintLimit: string
+  expectedSupply: string
+  inscriptionId: string
+}
+
+export function isOmigaInscription(udt: UDT): udt is OmigaInscription {
+  return 'mintStatus' in udt
```
Comment (Keith-CY) [inline src/models/UDT/index.ts:35]: It's prone to be covered by `sudt` or other token when they have `mint status` field for UI

File: src/pages/UDT/state.ts [hunk 3]
```diff
@@ -19,3 +19,11 @@ export const defaultUDTInfo: UDT = {
     hashType: '',
   },
 }
+
+export const defaultOmigaInscriptionInfo: OmigaInscription = {
+  ...defaultUDTInfo,
+  mintStatus: MintStatus.Closed,
+  mintLimit: '0',
+  expectedSupply: '0',
+  inscriptionId: '',
```
Comment (Keith-CY) [inline src/pages/UDT/state.ts:28]: `inscription id` is specific for `inscription item`(as implied in the document), naming it `id` or `inscriptionInfoId` would be clear

File: src/locales/zh.json [hunk 4]
```diff
@@ -581,7 +581,14 @@
...
       "address-or-hash": "交易哈希 / 地址",
-      "address": "地址"
+      "address": "地址",
+      "status": "状态",
+      "progress": "进度",
+      "expected_supply": "预期供应量",
+      "mint_limit": "铸造限制",
+      "mint_status_minting": "铸造中",
+      "mint_status_closed": "已关闭",
+      "mint_status_rebase_start": "重新基准开始"
```
Comment (Keith-CY) [inline src/locales/zh.json:591]:
```suggestion
      "mint_status_rebase_start": "重铸开始"
```
I've consulted with the project team and found `重铸` is better as the term

File: src/models/UDT/index.ts (hunk 1 above)
Comment (WhiteMinds) [reply src/models/UDT/index.ts]: Indeed, it has been updated.

File: src/models/UDT/index.ts (hunk 2 above)
Comment (WhiteMinds) [reply src/models/UDT/index.ts]: Expectedly, all UDT variants should have corresponding structures in the frontend, so there won't be incorrect judgments here.

However, a more comprehensive judgment would indeed be more robust. It has been updated.

File: src/locales/zh.json (hunk 4 above)
Comment (WhiteMinds) [reply src/locales/zh.json]: Updated.

File: src/pages/UDT/state.ts (hunk 3 above)
Comment (WhiteMinds) [reply src/pages/UDT/state.ts]: Updated.

File: src/components/Header/MenusComp/index.tsx [hunk 5]
```diff
@@ -59,29 +75,72 @@ const useMenuDataList = () => {
...
     </MobileMenuItem>
   ) : (
     <HeaderMenuPanel>
       {menuList.map(menu =>
-        menu.type === LinkType.Inner ? (
+        // eslint-disable-next-line no-nested-ternary
+        menu.children ? (
+          <SubmenuDropdown key={menu.name} menu={menu}>
+            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+            <a className={classNames('headerMenusItem', styles.submenuTrigger)}>
```
Comment (Keith-CY) [inline src/components/Header/MenusComp/index.tsx:138]: Why not use `Link` for navigation

**Review by Keith-CY — APPROVED** [review]

File: src/components/Header/MenusComp/index.tsx (hunk 5 above)
Comment (WhiteMinds) [reply src/components/Header/MenusComp/index.tsx:138]: This is a trigger, not a link. It doesn't have an href, it just uses an 'a' tag as a carrier. This structure is referenced from the antd example.

Comment (Keith-CY) [reply src/components/Header/MenusComp/index.tsx:138]: Got it

**Comment by FrederLu** [comment]
image
The style appears abnormal when the inscription info is displayed for too long.

**Comment by FrederLu** [comment]
image
The page should not appear blank when `Name` is empty.

---
image
image
There is a problem with the data display of `Holder Addresses`, `Issuer` and `Progress, Expected Supply, Mint Limit`.

**Comment by FrederLu** [comment]
https://ckb-explorer-frontend-in-magickbase-repo-o1sz84bhq-magickbase.vercel.app/sudt/0x88a2768a7eb6bb60892e7613e15647191762d8178595af24d2c1236738c76689
image
The progress bar displays abnormally after the additional issuance.

**Comment by FrederLu** [comment]
Inscription list page
API：
1、`24hr Transactions` takes a long time to update.
image
image

2、The 'address count' data should not all be 0.
image

3、The progress bar fills dependent data(`total_amount`) displays 0.     Compared with the `status` of the details page, it is slower to update from `minting` to `closed`, causing the data on both sides to be out of sync.
image

---
Inscription details page
API：
`Holder Addresses`, `Issuer` and `Progress` data problems still exist.
image

page：
The format exported by `CSV Export` is the previous token, and the export file is empty.
image

**Comment by FrederLu** [comment]
image

**Comment by FrederLu** [comment]
After filtering, `csv export` button will appear.

https://github.com/Magickbase/ckb-explorer-frontend/assets/25676576/cd3305a7-0949-4edd-89e5-97bc45aed91c

**Comment by WhiteMinds** [comment]
> After filtering, `csv export` button will appear.

Fixed, but the search will still redirect to the wrong address, which requires backend involvement to handle @zmcNotafraid .

**Comment by WhiteMinds** [comment]
> > After filtering, `csv export` button will appear.
> 
> Fixed, but the search will still redirect to the wrong address, which requires backend involvement to handle @zmcNotafraid .

@FrederLu Fixed, please check.

**Comment by WhiteMinds** [comment]
> There is an export button after filtering under the `Inscriptions` tab, but not under `sUDT`. The logic seems to be reversed.

Indeed, it has been fixed.

**Pull request merged**